runtime-sdk/modules/evm/src/signed_query.rs

```diff
@@ -0,0 +1,248 @@
+use std::convert::TryFrom as _;
+
+use ethabi::Token;
+use k256::{ecdsa::recoverable, elliptic_curve::sec1::ToEncodedPoint as _};
+use once_cell::sync::OnceCell;
+use sha3::{Digest as _, Keccak256};
+
+use oasis_runtime_sdk::{
+    context::Context, core::common::crypto::hash::Hash, modules::accounts::API as _,
+};
+
+use crate::{
+    state,
+    types::{Leash, SimulateCallQuery},
+    Config, Error,
+};
+
+pub(super) fn verify<C: Context, Cfg: Config>(
```

Comment:
Add documentation comments.

Also explain in the relevant comments why the eth ABI is used for some of these structures (I assume it is for compatibility with the eth personal signing scheme?).